src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -66,8 +71,143 @@ static std::string getDeviceFullName() {
+        };
+
+        brand_string = read_sysctl_str("machdep.cpu.brand_string");
+        if (brand_string.empty()) {
+            brand_string = read_sysctl_str("hw.model");
+        }
+    }
+#    elif defined(__linux__)
+    {
+        auto trim = [](std::string s) -> std::string {
+            const auto start = s.find_first_not_of(" \t\r\n");
+            const auto end = s.find_last_not_of(" \t\r\n");
+            if (start == std::string::npos || end == std::string::npos) {
+                return {};
+            }
+            return s.substr(start, end - start + 1);
+        };
+        auto read_first_line = [&](const char* path) -> std::string {
+            std::ifstream f(path);
+            if (!f.is_open()) {
+                return {};
+            }
+            std::string line;
+            std::getline(f, line);
+            return trim(line);
+        };
+        auto pick_value = [&](const std::string& s) -> std::string {
+            const auto pos = s.find(':');
+            if (pos == std::string::npos) {
+                return {};
+            }
+            return trim(s.substr(pos + 1));
+        };
+
+        // 1) Prefer device-tree model if available (not present in many containers)
+        brand_string = read_first_line("/sys/firmware/devicetree/base/model");
+        if (brand_string.empty()) {
+            brand_string = read_first_line("/proc/device-tree/model");
+        }
+
+        // 2) Fall back to /proc/cpuinfo keys commonly seen on ARM
+        if (brand_string.empty()) {
+            std::ifstream cpuinfo("/proc/cpuinfo");
+            std::string line;
+            std::string implementer_hex;  // e.g., 0x41
+            std::string part_hex;         // e.g., 0xd40
+            while (cpuinfo.is_open() && std::getline(cpuinfo, line)) {
+                if (line.rfind("model name", 0) == 0 || line.rfind("Hardware", 0) == 0 ||
+                    line.rfind("Processor", 0) == 0 || line.rfind("Model", 0) == 0) {
+                    auto v = pick_value(line);
+                    if (!v.empty()) {
+                        brand_string = v;
+                        break;
+                    }
+                } else if (line.rfind("CPU implementer", 0) == 0) {
+                    implementer_hex = pick_value(line);
+                } else if (line.rfind("CPU part", 0) == 0) {
+                    part_hex = pick_value(line);
+                }
+            }
+
+            // 3) If we still don't have a friendly string, synthesize something readable
+            if (brand_string.empty()) {
+                auto vendor_from_impl = [](const std::string& hex) -> const char* {
+                    // Map common implementer IDs (see Linux arch/arm64/include/asm/sysreg.h / MIDR)
+                    if (hex.length() >= 3) {
+                        auto id = std::stoul(hex, nullptr, 16);
+                        switch (id) {
+                        case 0x41:
+                            return "ARM";
```

Comment:
The vendor IDs are hardcoded magic numbers. Consider defining them as named constants:
  namespace ARM_VENDOR_ID {
      constexpr uint32_t ARM = 0x41;
      constexpr uint32_t BROADCOM = 0x42;
      // ...
  }

Reply:
Done